### DISCUSSION TITLE
Creating many to many relationships

### DISCUSSION THREAD

**Discussion by taylorpreston** [prisma/prisma#8486] [Q&A] [answered]
Hey guys! I think I am missing something here. Should this code work provided the schema? 

```ts
 const licenseRelationship = await prisma.licenseRelationship.create({
    data: {
      userId,
      licenseId
    }
  })
 ```

ERROR: `Foreign key constraint failed on the field: 'LicenseRelationship_userId_fkey (index)'`

```prisma
model User {
  createdAt         DateTime              @default(now()) @map(name: "created_at")
  email             String                @unique
  emailVerified     DateTime?
  firstName         String
  password          String
  id                String                @id @default(uuid())
  lastName          String
  supervised        SupervisionRelation[] @relation("supervised")
  supervisor        SupervisionRelation[] @relation("supervisor")
  updatedAt         DateTime              @updatedAt @map(name: "updated_at")
  sessions          Session[]
  Account           Account[]
  SupervisionInvite SupervisionInvite[]   @relation("sender")
  licenses          LicenseRelationship[]

  @@map(name: "users")
}

model License {
  createdAt    DateTime              @default(now()) @map(name: "created_at")
  formSchema   Json
  id           Int                   @id @default(autoincrement())
  requirements Json
  state        String
  type         String
  updatedAt    DateTime              @updatedAt @map(name: "updated_at")
  year         String
  users        LicenseRelationship[]

  @@map(name: "licenses")
}

model LicenseRelationship {
  createdAt DateTime @default(now()) @map(name: "created_at")
  license   License  @relation(fields: [licenseId], references: [id])
  licenseId Int
  user      User     @relation(fields: [userId], references: [id])
  userId    String

  @@id([userId, licenseId])
}

**Comment by ryands17** [comment]
@taylorpreston 👋 
Are the `userId` and `licenseId` both defined? If any one of those is undefined/null then the query will fail.

**Reply by taylorpreston** [reply]
They are both defined.😭

**Reply by ryands17** [reply]
Could you share the data you're passing or a sample repo? I cannot reproduce this when I ran this query on the latest version of Prisma.

**Reply by taylorpreston** [reply]
License
`
   {
      createdAt: '2021-07-28T00:29:17.542Z',
      formSchema: [Object],
      id: 1,
      requirements: [Object],
      state: 'TN',
      type: 'LCSW',
      updatedAt: '1970-01-01T00:00:00.000Z',
      year: '2021'
    }
   `
User
`{
  createdAt: '2021-07-26T16:33:16.820Z',
  email: 'testing123@soundstripe.com',
  emailVerified: null,
  firstName: 'Test',
  password: '$2a$10$Ey3j1ioCUtnuJDnERZ1ctOSrU1LD2FkLc2MabwyCwaBzL4JCM3.4i',
  id: '5d746bbe-e096-4918-82be-301af0b94860',
  lastName: 'Test',
  updatedAt: '2021-07-26T16:33:16.820Z'
}`

**Reply by taylorpreston** [reply]
🤘 Thanks for taking a look @ryands17

**Reply by taylorpreston** [reply]
I'm also using this with Next.js I don't know if there is a gotcha there.

**Comment by taylorpreston** [comment, marked as answer]
🤦 

So I haven't set up a way to destroy a session if the user doesn't exist in the database. I was getting an old user that was deleted from the database. So of course it wasn't working because the user didn't exist. 😭 Haha

**Comment by taylorpreston** [comment]
@ryands17 Thanks for taking a look man. I appreciate the help even though it was me being a dummy.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
